### ISSUE TITLE
Trove: The behavior of `AttachToInstance` differs from what is documented

### ISSUE THREAD

**Issue by lolmanurfunny** [Sleitnick/RbxUtil#147]
The `:AttachToInstance()` method listens to `Instance.Destroying`, this is no doubt unintended as the docs imply that it uses `Instance.AncestryChanged`.

The doc's description: *"... Once this instance is removed from the game (**parent or ancestor's parent set to nil**), the trove will automatically clean up."*

However, due to the nature of the `Instance.Destroying` event, the trove will not clean up unless the instance's `:Destroy()` method was explicitly called.
- - -
https://github.com/Sleitnick/RbxUtil/blob/main/modules/trove/init.lua#L370

**Comment by lolmanurfunny** [comment]
> > However, due to the nature of the Instance.Destroying event, the trove will not clean up unless the instance's :Destroy() method was explicitly called.
> 
> This is not entirely true, the event can be internally fired too if the instance is destroyed through other means by the engine (such as an unanchored part falling off the World where `position.Y` >= `Workspace.FallenPartsDestroyHeight`.

Are you sure this is the case? I attempted to reproduce this behavior as you described, in Studio. It appears that you are in fact completely wrong, that's putting it lightly.

https://github.com/Sleitnick/RbxUtil/assets/77128366/deaee958-228f-4258-b6e9-1be5d27f7a29

**Comment by bubshayz** [comment]
![image](https://github.com/Sleitnick/RbxUtil/assets/92627259/0e85f414-6ac2-47be-99fa-e6d7213b1df6)

@lolmanurfunny  https://devforum.roblox.com/t/instancedestroy-only-replicates-parent-nil-change/127327/14

**Comment by Sleitnick** [comment]
Dealing with Roblox instances can be frustrating. If the ancestor of an instance is set to `nil`, it can be hard to know if this was done by the engine or by a script. I think the underlying code here should probably be switched back to AncestryChanged (which is what it used to be).

To be clear, I think `AttachToInstance` probably shouldn't exist on Trove. It inverts the ownership. Ideally, code that wants to clean up a trove based on an instance should write code to bind it to the instance themselves, thus giving the developer the ability to choose between ancestry change or destroy.

**Comment by lolmanurfunny** [comment]
> To be clear, I think `AttachToInstance` probably shouldn't exist on Trove. It inverts the ownership. Ideally, code that wants to clean up a trove based on an instance should write code to bind it to the instance themselves, thus giving the developer the ability to choose between ancestry change or destroy.

I agree. I've never needed to utilize `AttachToInstance` myself since IMO it typically makes more sense to use a kind of lifecycle (eg: Death) to cleanup troves. IME, instance deletion is the *effect* of trove cleanup more often than it is the *cause*.
